src/main/scala/org/constellation/gossip/sampling/PartitionerPeerSampling.scala

```diff
@@ -14,25 +14,32 @@ import org.constellation.util.Partitioner._
 
 class PartitionerPeerSampling[F[_]: Concurrent](selfId: Id, cluster: Cluster[F], trustManager: TrustManager[F])
     extends PeerSampling[F] {
-  private val partitionCache: Ref[F, List[GossipPath]] = Ref.unsafe(List.empty[GossipPath])
+  type Partition = IndexedSeq[Id]
+
+  private val partitionCache: Ref[F, List[Partition]] = Ref.unsafe(List.empty[Partition])
   private val trustDataInternalCache: Ref[F, List[TrustDataInternal]] = Ref.unsafe(List.empty[TrustDataInternal])
 
   private val logger: SelfAwareStructuredLogger[F] = Slf4jLogger.getLogger[F]
 
   override def selectPaths: F[List[GossipPath]] =
     cachedPartitionsCoverAllPeers.ifM(
-      logger.debug("Cached partitions cover all the peers. Returning.") >> partitionCache.get,
+      logger.debug("Cached partitions cover all the peers. Returning.") >> partitionCache.get
+        .flatMap(partitionsToGossipPaths),
       logger
         .debug("Some nodes are missing in cached partitions. Repartitioning.") >> repartitionWithDefaults() >> partitionCache.get
+        .flatMap(partitionsToGossipPaths)
     )
 
   def repartition(selfTdi: TrustDataInternal, tdi: List[TrustDataInternal]): F[Unit] =
     for {
-      partitions <- calculatePartitions(selfTdi, tdi)
+      partitions <- calculatePartitions(selfTdi, tdi).pure[F]
```

Comment:
is calculation pure? shouldn't it be `F.delay`?

Reply:
or actually if it is not pure, let's wrap the output type of this fn with `F[.]`

Reply:
The calculation is pure. Previously the result type of calculation was wrapped in `F[_]` because it was generating random UUID. Now the random UUID is no longer calculated on repartition (nor cached).